src/tests/desktop-agent.spec.js

```diff
@@ -54,18 +83,56 @@ describe("Desktop Agent", function() {
       it("is defined", function() {
         expect(desktopAgent.findIntent).toBeDefined();
       });
+      it("returns an AppIntent object", function() {
+        return desktopAgent.findIntent("StartChat").then(function(result) {
+          expect(result).toBeDefined();
+          expect(result.intent).toBeDefined();
+          expect(result.apps).toBeDefined();
+        });
+      });
     });
 
     describe("findIntentsByContext()", function() {
       it("is defined", function() {
         expect(desktopAgent.findIntentsByContext).toBeDefined();
       });
+      it("returns an array of AppIntent objects", function() {
+        let context = {};
+        return desktopAgent.findIntentsByContext(context).then(function(result) {
+          expect(result).toBeDefined();
+          result.forEach(val => {
+            expect(val.intent).toBeDefined();
+            expect(val.apps).toBeDefined();
+          });
+        });
+      });
     });
 
     describe("raiseIntent()", function() {
       it("is defined", function() {
         expect(desktopAgent.raiseIntent).toBeDefined();
       });
+
+      it("returns an IntentResolution object", function() {
+        let context = {};
+        return desktopAgent.raiseIntent("StartChat", context).then(function(result) {
+          expect(result).toBeDefined();
+          expect(result.source).toBeDefined();
+          expect(result.version).toBeDefined();
+        });
+      });
+
+      it("returns an error if app was not found", function() {
```

Comment:
I don't know if this test is correct. The documentation says that "If the resolution fails, the promise will return an Error with a string from the [ResolveError](https://fdc3.finos.org/docs/1.0/api/Errors#resolveerror) enumeration."

I would expect a `"NoAppsFound"` string returned because that would be the relevant string from the ResolveError enum

Reply:
I swear the documentation didn't say that when I looked at it...must have been an oversight on my part. Will update accordingly